File: control-plane/internal/domain/entity/system.go
package entity

// Tuân thủ Flat Entity: không chứa json tags.

// SystemInfo là flat projection chứa toàn bộ thông tin trạng thái hoạt động của hệ thống.
type SystemInfo struct {
	Product              string
	Version              string
	Build                string
	GoVersion            string
	UptimeSeconds        int64
	UptimeFormatted      string
	Architecture         string
	StatePersistence     string
	DatabasePath         string
	DatabaseSizeBytes    int64
	DatabaseSizeFormat   string
	MemoryAllocBytes     uint64
	MemoryAllocFormatted string
}

// ControllerStatus là flat entity chứa trạng thái hoạt động của controller.
type ControllerStatus struct {
	Component        string
	Stage            string
	EnforcementReady *bool
	Message          string
}

File: ui/src/lib/api/system.ts
import { api } from '../fetcher';

export interface SystemInfo {
  product: string;
  version: string;
  build: string;
  go_version: string;
  uptime_seconds: number;
  uptime_formatted: string;
  architecture: string;
  state_persistence: string;
  database_path: string;
  database_size_bytes: number;
  database_size_formatted: string;
  memory_alloc_bytes: number;
  memory_alloc_formatted: string;
}

export const systemApi = {
  getInfo: () => api.get<SystemInfo>('/api/v1/system/info'),
};
